packages/client/src/components/button/button.tsx

```diff
@@ -0,0 +1,16 @@
+import styles from './button.module.css';
+
+
+export const Button = (props: React.ComponentPropsWithoutRef<'button'> & {mode?: 'default' | 'link' | 'danger'}) => {
+  const { children = 'Кнопка', className, mode = 'default'} = props;
+
+  const classes = className ? `${styles.button} ${styles['button--'+mode]} ${className}` : `${styles.button} ${styles['button--'+mode]}`;
+
+  return (
+    <button {...props} className={classes}>
+      {children}
+    </button>
+  )
+}
+
+export default Button;
```

Comment:
`Запрос корректировки`
Предлагаю остановится на одной единой для всех и универсальной реализации компонента [Button](https://github.com/Aleksandr-86/canvas-bomberman/blob/af-23-add-landing-page/packages/client/src/components/button/button.tsx)
```suggestion
export const Button = (props: React.ComponentPropsWithoutRef<'button'>) => {
  const { children = 'Кнопка', className } = props

  const classes = className ? `${styles.button} ${className}` : styles.button

  return (
    <button {...props} className={classes}>
      {children}
    </button>
  )
}
```

Reply:
Может тогда стоит отделить стиль кнопки от стилей в пропсах? В таком случае получится более тонко настраивать стили для кнопки. По шаблону у нас их три типа, а в текущей реализации получается, что нельзя отказаться от изменения background'a у кнопки, т.к он прописан в стиле по умолчанию, и его нельзя не использовать. Или стоит сделать другой компонент для этого?
```suggestion
export const Button = (props: React.ComponentPropsWithoutRef<'button'>) => {
  const { children = 'Кнопка', className } = props
  const classes = className ?  styles.className : styles.button
  return (
    <button {...props} className={classes}>
      {children}
    </button>
  )
}
```

Reply:
> Может тогда стоит отделить стиль кнопки от стилей в пропсах?  а в текущей реализации получается, что нельзя отказаться от изменения background'a у кнопки, т.к он прописан в стиле по умолчанию, и его нельзя не использовать. Или стоит сделать другой компонент для этого?

> Может тогда стоит отделить стиль кнопки от стилей в пропсах? В таком случае получится более тонко настраивать стили для кнопки. 

Можно. Но в таком случае нам не нужен компонент `Button` верно? Зачем нам пустая обёртка, которая ничего не делает, то есть не применяет какие то значения по умолчанию?

> По шаблону у нас их три типа  

Я в прототипе вижу только один тип кнопки))
![image](https://user-images.githubusercontent.com/96790009/212398991-11714102-12e8-4564-85ed-3ddb1dabbd25.png)
Какие ещё два типа видишь ты?

> а в текущей реализации получается, что нельзя отказаться от изменения background'a у кнопки, т.к он прописан в стиле по умолчанию  

Странное дело. Я хорошо помню, что я тестировал кнопку на переопределение стилей. Сейчас нашёл устаревшую версию в архиве. Там всё прекрасно переопределяется. Вот так к примеру работает изменение фона при наведении на кнопку курсора:
![image](https://user-images.githubusercontent.com/96790009/212406818-dc228df7-88af-4535-9c60-11667db0cf4b.png)
Пытаюсь сейчас понять в чём дело.

>  Или стоит сделать другой компонент для этого?
Ну не знаю. Идея компонента `Button` в том, чтобы использовать один и тот же стиль кнопки, с возможностью изменения стилей. Например на форуме возможно понадобятся кнопки, меньшего размера, с меньшим шрифтом и границами.
Я переделал страницы, уйдя от вложенных в компоненты `Link` кнопок, к обычным ссылкам со стилизацией под кнопку. Если мы сможем договориться, то можно вообще не использовать отдельный компонент `Button`, стилизуя сам DOM элемент `button` одинаковым образом.

Reply:
Стили изменяются, если использовать кнопку на одной из пустых страниц. 🤔

Reply:
Разобрался. Импорт компонента `Button` не должен быть после импорта стилей. Вероятно у тебя была схожая со мной ситуация, когда ты тестируя компонент `Button` импортировал его "на быструю" через выпадающее меню ПКМ, а после не позаботился о его положении среди других импортов.

Reply:
Под типы я выделил еще такую реализацию

![image](https://user-images.githubusercontent.com/109624658/212426869-218b4190-5095-469a-ae1d-8b83fce8a358.png)

Здесь нет рамки и фона. Или я что-то путаю и здесь не компонент Button?

Reply:
Ну, не то чтобы путаешь. Это могли бы быть кнопки, если бы не осуществлялся переход на другую страницу.
Обрати внимание на блок схему:
![image](https://user-images.githubusercontent.com/96790009/212433180-ef795664-f0c0-4c90-9dbb-0bbd8f78c053.png)
При нажатии на `изменить пароль` или `выйти` пользователь должен быть перенаправлен на страницу изменения пароля или посадочную страницу. То есть здесь вполне можно обойтись ссылками `Link` импортированными из react-router-dom и стилизованными под кнопку. А вот для кнопки обработку нажатия `Сохранить` вероятно лучше делать через кнопку (см. блок схему), так как мы перед перенаправлением должны проверить соответствие содержания полей формы установленным правилам. Если всё нормально, перенаправляем пользователя. Если нет, выводим информацию о том, какие правила он нарушил.

Reply:
Понял, спасибо.